std/typecons.d

```diff
@@ -1732,8 +1761,36 @@ succeeds, $(D this) becomes non-null.
     }
 
 /**
+    If this `Nullable` wraps a type that already has a null value
+    (such as a pointer), then assigning the null value to this
+    `Nullable` is no different than assigning any other value of
+    type `T`, and the resulting code will look very strange. It 
+    is strongly recommended that this be avoided by instead using
+    the version of `Nullable` that takes an additional `nullValue`
+    template argument.
+ */
+unittest
+{
+    //Passes
+    Nullable!(int*) npi;
+    assert(npi.isNull);
+    
+    //Passes?!
```

Comment:
I strongly argued for that in #2593 but met a lot of resistance for various reasons. I stand by my arguments from then.

If we don't fix important type constructors like `RefCounted`, `Unique` and `Nullable` we'll keep getting flak from C++ enthusiasts who use these implementations as arguments against D's type system.